scene/resources/particles_material.h

```diff
@@ -92,7 +92,7 @@ class ParticlesMaterial : public Material {
 			uint32_t texture_mask : 16;
 			uint32_t texture_color : 1;
 			uint32_t particle_flags : 4;
-			uint32_t emission_shape : 2;
+			uint32_t emission_shape : 3;
```

Comment:
I'd make it 4 to be safe and make the sum 32 instead of 31.

Reply:
Which is the size of `key` in the union too, so it should keep it all on 32-bit IINM.

Reply:
Alternatively, we can use 3 here and add a 1 bit padding field to make it 32. I assume that was the intent of this union, but maybe some fields were removed, make the size 30 bits.

Reply:
I've made it to 4 in both `master` and `3.x` versions. Since it's all for padding, I guess it wouldn't matter until there are more fields in that struct and we need to reconsider the bits. In `3.x` it doesn't result in a pretty round number like 32 though 🙁

Reply:
Yeah I've asked reduz and clayjohn for feedback on RC, maybe 3 was indeed better and we should only use what we need to leave room for more fields (though in `master` we've almost exhausted all bits).

Reply:
@jfons says:

> as long as they are kept under or equal 32 bits it should be fine

Let's go back to `3` since that's what needed, so we keep one bit free. And add a comment before the struct to make it explicit that the sum of fields should be kept <= 32.

Might also be worth adding a comment before the emission shape enum definition to make contributors aware of that restriction if they had several new shapes.

Reply:
Done, added comments to other participating enums too.